samples/file-encryptor/enclave/openssl/encryptor.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "encryptor.h"
+#include <string.h>
+#include "../common/common.h"
+
+ecall_dispatcher::ecall_dispatcher() : m_encrypt(true), m_header(NULL)
+{
+    unsigned char iv[IV_SIZE] = {0xb2,
```

Comment:
Apparently I did not review the original sample when it was merged, or I really should have blocked this. It's unsafe to be using a fixed IV with the same key for AES and we should definitely not be showcasing that in a sample. At the very least, we need to very clearly call out that this is a bad practice.

Since the encryptor API is set up to manage the key and IV generation independent of the caller, the best approach here is probably to generate a random IV, and to store that IV with the ciphertext. There may still be concerns of random collisions reducing the usable space of IV values, but that's still way better than a fixed constant IV.

Reply:
@mingweishih can you actually do a review of the existing sample and evaluate the use of crypto? This worries me a little given the tendency of devs to copy and paste sample code.

Reply:
Sure.

Reply:
Created https://github.com/openenclave/openenclave/issues/3692 to track this